src/EnergyPlus/Coils/CoilCoolingDXCurveFitOperatingMode.cc

```diff
@@ -298,7 +298,8 @@ void CoilCoolingDXCurveFitOperatingMode::CalcOperatingMode(EnergyPlus::EnergyPlu
     // Currently speedNum is 1-based, while this->speeds are zero-based
     auto &thisspeed(this->speeds[max(speedNum - 1, 0)]);
 
-    if ((speedNum == 0) || ((speedNum == 1) && (speedRatio == 0.0)) || (inletNode.MassFlowRate == 0.0)) {
+    if ((speedNum == 0) || ((speedNum == 1) && (speedRatio == 0.0)) || (inletNode.MassFlowRate == 0.0) ||
+        (this->coilCoolingDXAvailSched->getCurrentVal() <= 0.0)) {
```

Comment:
If the DX cooling coil is scheduled off then the operating mode calculate function turns off the coil and sets the operating variables accordingly. This is the only place where the coil is turned off. Note the availability schedule pointer is propagated to all the operating modes such that each operating mode is controlled based on the schedule specified in the parent `Coil:Cooling:DX` object.